internal/beater/beater.go

```diff
@@ -379,8 +379,16 @@ func (s *Runner) Run(ctx context.Context) error {
 		finalBatchProcessor,
 	}
 
+	agentConfigFetcher, err := newAgentConfigFetcher(ctx, s.config, kibanaClient, newElasticsearchClient)
+	if err != nil {
+		return err
+	}
+	g.Go(func() error {
+		return agentConfigFetcher.Run(ctx)
+	})
```

Comment:
Rather than adding Run to the agentcfg.Fetcher interface, can we either pass the errgroup into newAgentConfigFetcher, or have that function return a separate run function? Then we can remove all of those no-op/passthrough Run methods.

Reply:
I prefer `beater` to be responsible of starting goroutines if possible and `newAgentConfigFetcher` to not run anything, since it is just another "new" function. Changed the function to have an extra return value for the run function.

Reply:
Thanks, sounds good to me.